### PR TITLE
CLI update to cater for request ACL within Bundle endpoints

### PR DESCRIPTION
**Pull request by fsat**
- Implement `conduct acls` command to display request ACL within Bundle endpoints
- Update `conduct services` command to only include Bundle having endpoint declared with service URI

### CODE REVIEW

**Comment by huntc** [comment]
We will however still require a `services` command... it will present the ACLs right?

**Comment by fsat** [comment]
At the moment, I haven't thought about how ACLs will be represented, hence the WIP. At least with this change, the `conduct services` won't error if `services` field is not present.

**Comment by fsat** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;kXdFMJr0&#x2F;174-conductr-haproxy-template-configuration"> conductr-haproxy template configuration</a>

**Comment by fsat** [comment]
Implemented `conduct acls` command which accepts either `http` or `tcp` as input argument.

**Comment by fsat** [comment]
Manual test performed successfully.

I have a test bundle called `path-tester` which exposes the following ACL:

```
BundleKeys.endpoints := Map(
  "ptest" -> Endpoint("http", 0, "path-tester",
    RequestAcl(
      Http(
        "^/bacon".r -> "/burger",
        "^/tree".r,
        "^/fee/(.*)/fi/(.*)/fo/(.*)/fum$".r -> "/boom/\\1-\\2-\\3/box",
        "^/boom/(.*)/box$".r,
        "/foo" -> "/baz",
        "/baz"
      )
    )
  ),
  "ptunnel" -> Endpoint("tcp", 0, "ptunnel",
    RequestAcl(
      Tcp(
        12101, 3303, 5601
      )
    )
  )
)
```

I have loaded the `path-tester` bundle into sandbox started with `sandbox run 0.1.0 -f logging -f visualization -p 9200`.

The ACL information span across these bundles:

```
Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct info
ID               NAME                    #REP  #STR  #RUN
d687c32          conductr-elasticsearch     1     0     1
1b2f457-9c7984a  conductr-haproxy           1     0     1
468e4dd          conductr-kibana            1     0     1
bd41633          path-tester                1     0     1
a6ae382          visualizer                 1     0     1
```

Calling `conduct acls http` displays the following

```
Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct acls http
METHOD  PATH                             REWRITE             BUNDLE ID  BUNDLE NAME             STATUS
*       ^/fee/(.*)/fi/(.*)/fo/(.*)/fum$  /boom/\1-\2-\3/box  bd41633    path-tester             Running
*       ^/boom/(.*)/box$                                     bd41633    path-tester             Running
*       ^/elastic-search                 /                   d687c32    conductr-elasticsearch  Running
*       ^/                                                   468e4dd    conductr-kibana         Running
*       ^/bacon                          /burger             bd41633    path-tester             Running
*       ^/tree                                               bd41633    path-tester             Running
*       ^/                                                   a6ae382    visualizer              Running
*       /foo                             /baz                bd41633    path-tester             Running
*       /baz                                                 bd41633    path-tester             Running
```

Calling `conduct acls tcp` displays the following

```
Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct acls tcp
TCP/PORT  BUNDLE ID  BUNDLE NAME             STATUS
3303      bd41633    path-tester             Running
5601      bd41633    path-tester             Running
9300      d687c32    conductr-elasticsearch  Running
12101     bd41633    path-tester             Running
```

File: conductr_cli/conduct_acls.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    ] + sorted([
+        {
+            'tcp_port': tcp_port,
+            'bundle_id': tcp_acl['bundle_id'],
+            'bundle_name': tcp_acl['bundle_name'],
+            'status': tcp_acl['status']
+        }
+        for tcp_acl in tcp_acls
+        for tcp_port in tcp_acl['acl']['tcp']['requests']
+    ], key=tcp_port_value)
```
Comment (fsat) [inline conductr_cli/conduct_acls.py:73]: Ordering of the TCP based ACL will be displayed according to what ConductR's ordering, i.e. sorted by port number.

File: conductr_cli/conduct_acls.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    ], key=path_depth_reverse)
+
+    data = [{
+        'http_method': 'METHOD',
+        'http_acl': 'PATH',
+        'http_rewrite': 'REWRITE',
+        'bundle_id': 'BUNDLE ID',
+        'bundle_name': 'BUNDLE NAME',
+        'status': 'STATUS'
+    }] + http_path_regex_sorted + http_path_beg_sorted + http_path_sorted
```
Comment (fsat) [inline conductr_cli/conduct_acls.py:128]: Ordering of the HTTP based ACL will be displayed according to what ConductR's ordering:
- Path Regex - within path regex, it's sorted by the number of capture group in reverse
- Path Begin - within path begin, it's sorted by path depth in reverse
- Path - within path, it's sorted by path depth in reverse

File: conductr_cli/conduct_acls.py (hunk 1 above)
Comment (huntc) [inline conductr_cli/conduct_acls.py:73]: Nice

File: conductr_cli/conduct_acls.py (hunk 2 above)
Comment (huntc) [inline conductr_cli/conduct_acls.py:128]: Nice

**Comment by huntc** [comment]
Nice work! Can we also update sbt-conductr so that it includes the new options?

**Comment by fsat** [comment]
Thank you, and yes - I shall do so.

**Pull request merged**